### DISCUSSION TITLE
Support `npm` and `local>` syntax with one config?

### DISCUSSION THREAD

**Discussion by fullstackzach** [renovatebot/renovate#9877] [Request Help] [answered]
Hello! I am trying to transition our repos off of the [npm hosted presets](https://docs.renovatebot.com/config-presets/#npm-hosted-presets) pattern, which are now deprecated, to a [`local>` hosted presets](https://docs.renovatebot.com/config-presets/#local-presets) pattern. 

I'm trying to figure out a solution where one config can support both patterns, since it will take some time for individual repos to switch. Ideally I'd like to avoid duplicating the config and would prefer both patterns to refer to the same config.  It looks like npm expects configs to live in the package.json under a "renovate-config" object. `local>` expects to search `default.json` first, and then `renovate.json`.  Is it possible to have `local>` reference the package.json or vice versa - npm config reference default.json

What do you recommend as a best pattern for the renovate-config shared preset repo during this transition?

A side effect I have also found is, our org renovate-config repo which hosts the preset, already had a renovate.json that referred to itself so it could be renovated. (eg. `{ extends: ['@org/renovate-config'] }`). This causes repos using the `local>` config to open an issue saying, `Error type: Cannot find preset's package (@org/renovate-config). Note: this is a nested preset so please contact the preset author if you are unable to fix it yourself.`. .. Which is somewhat confusing.

Thanks! I really appreciate the time you take to think about issues and configuration questions!

**Comment by viceice** [comment, marked as answer]
Your can use `default.json` for shared preset, as it's preferred over `renovate.json`.

I've done such a migration too.

1. Create the shared local presets.
2. Publish new npm presets to extend the new local presets.
3. Update repos to use new local presets.

**Reply by viceice** [reply]
best to move the `renovate.json` from shared repo to `.github` folder (works on any platform. 😉

 There are some more platform folders you can use.

**Reply by kreig303** [reply]
@viceice a bit confused... do you know of an example repo which shows this "fusion" pattern?

**Reply by viceice** [reply]
No public, sorry. Which platform are you using? Are your shared presets public?

**Reply by kreig303** [reply]
self-hosted gitlab. corporate so no, not public 😅

**Reply by viceice** [reply]
Ok, do you have a single shared preset or multiple?

**Reply by kreig303** [reply]
multiple, with hierarchies

```
preset a 
|_> preset b
  |_> repo-specific renovate.json
```

**Reply by kreig303** [reply]
gonna hash this out with an example i found... will let you know  😉

**Reply by viceice** [reply]
https://docs.renovatebot.com/config-presets/

Did you checked the docs recently? We've done some improvement recently.

**Reply by fullstackzach** [reply]
Hey @viceice, thanks for helping out. I am trying to imagine the file structure that would work in practice, something like this?

renovate-config repo
```
package.json // to support npm
default.json // to support >local
```

package.json
```
{
  "renovate-config": {
     "extends": ['local>org/renovate-config']
  }
}
```

default.json
```
{
   .. existing shared renovate config ..
}
```

**Reply by viceice** [reply]
Yes, that works for default preset. 👍

Here a sample from me, with some preset hosted on GitHub, can be used as local on GitHub.

https://github.com/visualon/renovate-config

**Reply by fullstackzach** [reply]
This scheme worked! I was able to get the npm and local> patterns working referencing one central preset. https://github.com/renovatebot/renovate/discussions/9877#discussioncomment-700763
Thanks for talking it out.
